lib/internal/modules/cjs/loader.js

```diff
@@ -764,9 +764,15 @@ Module.runMain = function() {
 };
 
 Module.createRequireFromPath = (filename) => {
-  const m = new Module(filename);
-  m.filename = filename;
-  m.paths = Module._nodeModulePaths(path.dirname(filename));
+  // Allow a directory to be passed as the filename
+  const proxyPath = filename.endsWith('/') ?
+    path.join(filename, 'noop.js') :
```

Comment:
👆 Windows filepath could end with a backslash.

Reply:
Should we test for either? Or could we use `path.sep`?

Reply:
You can use other places (like /lib/path) for reference. When dealing with `win32` either forward or backward slash is allowed.

Reply:
Thanks for the pointer, I'll have a look :)

Reply:
Since `path.win32.normalize` will always resolve to a windows-style path on either platform we might as well use it and check for a `CHAR_BACKWARD_SLASH` 👍

Reply:
It's a shame a dummy filename is needed here.

An alternative might be to handle the trailing slash in the `path.dirname` calls of lib/internal/modules/cjs/loader.js`, although it would have to be done carefully then not to pollute the logic.

Reply:
I honestly didn't want to mess with the logic in `lib/internal/modules/cjs/loader.js`, it could be a can of worms.

Reply:
The filename is only accessed in two places for its directory - https://github.com/nodejs/node/blob/master/lib/internal/modules/cjs/loader.js#L427 and https://github.com/nodejs/node/blob/master/lib/internal/modules/cjs/loader.js#L496 so perhaps just a couple of worms.

Reply:
Marking as resolved.